lib/ui/src/modules/ui/components/left_panel/stories_tree/tree_decorators.js

```diff
@@ -84,22 +84,60 @@ ContainerDecorator.propTypes = {
   onClick: PropTypes.func.isRequired,
 };
 
-function HeaderDecorator(props) {
-  const { style, node } = props;
-
-  let newStyle = style;
-
-  if (node.type === treeNodeTypes.STORY) {
-    newStyle = {
-      ...style,
-      title: {
-        ...style.title,
-        ...style.storyTitle,
-      },
-    };
+class HeaderDecorator extends React.Component {
+  decorateNameMatchedToSearchTerm(node, style) {
+    const { storyFilter, name } = node;
+
+    if (!storyFilter) {
+      return name;
+    }
+
+    const filterRegex = new RegExp(`(${storyFilter})`, 'i');
```

Comment:
Looks like this will break if with special characters. I'd suggest to use [Fuse.js](fusejs.io) with `{ includeMatches: true }` in `libs/filters.js` and store the matches information in some spesial field on the filtered stories